mbf_costmap_nav/src/mbf_costmap_nav/costmap_navigation_server.cpp

```diff
@@ -102,8 +106,9 @@ CostmapNavigationServer::CostmapNavigationServer(const TFPtr &tf_listener_ptr) :
   speedLimiterExternalConfigServer_ = boost::make_shared<dynamic_reconfigure::Server<base_local_planner::ExternalSpeedLimiterConfig>>(speedLimiterExternal_nh);
   ros::NodeHandle speedLimiterShadow_nh("~/speed_limiters/shadow");
   speedLimiterShadowConfigServer_ = boost::make_shared<dynamic_reconfigure::Server<base_local_planner::ShadowSpeedLimiterConfig>>(speedLimiterShadow_nh);
-  
-  publishReadySignal(true);
+  map_chunk_id_subscriber_ = private_nh_.subscribe("/map_server/update_map_chunk", 1, &CostmapNavigationServer::updateMapChunkCB, this);
+
+  timer_ = private_nh_.createTimer(ros::Duration(1), &CostmapNavigationServer::publishReadyCB, this);
```

Comment:
should we stop the timer after the ready signal is published, and start when a map update message is recieved, so we aren't always checking during navigation on the same map chunk?

Reply:
yes, I'll add that.